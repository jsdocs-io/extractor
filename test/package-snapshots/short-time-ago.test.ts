import { describe, it } from 'vitest';
import { expectPackageAPIToMatchSnapshot } from '../helpers/expect-package-api-to-match-snapshot';

describe('short-time-ago', () => {
    const name = 'short-time-ago';

    it('2.0.0', async () => {
        await expectPackageAPIToMatchSnapshot({ name, version: '2.0.0' });
    });
});
