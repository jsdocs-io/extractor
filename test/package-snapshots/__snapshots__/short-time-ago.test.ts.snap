// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`short-time-ago 2.0.0 1`] = `
{
  "declarations": {
    "classes": [],
    "enums": [],
    "functions": [
      {
        "docs": [
          "/**
 * \`timeAgo\` returns a string describing the time elapsed between
 * a given date and the current time at which the function is called.
 *
 * @remarks
 * \`timeAgo\` only supports the \`en_US\` locale.
 *
 * The following table describes \`timeAgo\`'s output.
 *
 * \`\`\`
 * | Time elapsed          | Past output       | Future output    |
 * | --------------------- | ----------------- | ---------------- |
 * | < 1 second            | \`just now\`        | \`just now\`       |
 * | < 1 minute            | \`N second(s) ago\` | \`in N second(s)\` |
 * | < 1 hour              | \`N minute(s) ago\` | \`in N minute(s)\` |
 * | < 1 day               | \`N hour(s) ago\`   | \`in N hour(s)\`   |
 * | < 1 month (30.5 days) | \`N day(s) ago\`    | \`in N day(s)\`    |
 * | < 1 year (365 days)   | \`N month(s) ago\`  | \`in N month(s)\`  |
 * | > 1 year              | \`N year(s) ago\`   | \`in N year(s)\`   |
 * \`\`\`
 *
 * @example
 * Basic usage:
 *
 * \`\`\`typescript
 * import { timeAgo } from 'short-time-ago';
 *
 * const myDate = new Date();
 * const description = timeAgo(myDate);
 *
 * // Output: \`just now\`.
 * console.log(description);
 * \`\`\`
 *
 * @example
 * Specifying a custom current date with the \`now\` parameter:
 *
 * \`\`\`typescript
 * import { timeAgo } from 'short-time-ago';
 *
 * const myDate = new Date('2019-01-01T00:00:00.000Z');
 * const now = new Date('2019-01-01T00:01:00.000Z');
 * const description = timeAgo(myDate, now);
 *
 * // Output: \`1 minute ago\`.
 * console.log(description);
 * \`\`\`
 *
 * \`\`\`typescript
 * import { timeAgo } from 'short-time-ago';
 *
 * const myDate = new Date('2019-01-02T00:00:00.000Z');
 * const now = new Date('2019-01-01T00:00:00.000Z');
 * const description = timeAgo(myDate, now);
 *
 * // Output: \`in 1 day\`.
 * console.log(description);
 * \`\`\`
 *
 * @param date - a {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date | Date}
 * @param now - the current date (optional, defaults to \`new Date()\`)
 */",
        ],
        "id": "timeAgo",
        "kind": "FunctionDeclaration",
        "name": "timeAgo",
        "signature": "timeAgo: (date: Date, now?: Date) => string;",
        "source": {
          "filename": "src/index.ts",
          "line": 71,
          "unpkgURL": "https://unpkg.com/browse/short-time-ago@2.0.0/src/index.ts#L71",
          "url": "https://github.com/velut/node-short-time-ago/tree/v2.0.0/src/index.ts#L71",
        },
        "type": "(date: Date, now?: Date) => string",
      },
    ],
    "interfaces": [],
    "namespaces": [],
    "typeAliases": [],
    "variables": [],
  },
  "files": [
    {
      "filename": "src/index.ts",
      "isIndexFile": true,
      "unpkgURL": "https://unpkg.com/browse/short-time-ago@2.0.0/src/index.ts",
      "url": "https://github.com/velut/node-short-time-ago/tree/v2.0.0/src/index.ts",
    },
  ],
  "overview": "/**
 * This package exports a single function, {@link timeAgo},
 * which describes the time elapsed between a given date and the current date
 * in a human readable format (for example, \`10 minutes ago\`, \`in 3 seconds\`).
 *
 * @packageDocumentation
 */",
}
`;
